### ISSUE TITLE
memory leak in astparser module

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#3344]
ex:
```
david@greybeard:~/src/qore/git/qore/build$ ps axl | grep -e VSZ -e 36543
  UID   PID  PPID CPU PRI NI      VSZ    RSS WCHAN  STAT   TT       TIME COMMAND
  501 36543 26528   0  31  0 15339124      4 -      S      ??   21:18.41 qore /Users/david/.vscode/extensions/qoretechnologies.qore-vscode-0.3.3/qls/qls.q
```

the equivalent of PSS (Proportional Set Size) was > 9GB

source unknown - mem leak in qore binary in some specific API?  bad memory mgmt in qls.q?  Needs investigating

**Comment by omusil24** [comment]
Do you have any example files where this happens?

**Comment by davidnich** [comment]
@omusil24 no I'm sorry, I don't know - I have 4 VSCode windows open - 2 with very large projects (Qore & Qorus) and I don't know which process the `qls.q` binary in question corresponds to

**Comment by omusil24** [comment]
Fixed in #3579 and #3580
